lib/tic_tac_toe_web/controllers/board_controller.ex

```diff
@@ -1,19 +1,52 @@
 defmodule TicTacToeWeb.BoardController do
     use TicTacToeWeb, :controller
-  
-    def post_move(connection, _params) do
+    
+    alias TicTacToe.Game, as: Game
+
+    def new_game(conn, _params) do
+        game = Game.setup_new_game
+        game_in_map_form = Map.from_struct(game)
+        body = Jason.encode!(game_in_map_form)
+
+        conn
+        |> put_status(200)
+        |> text(body)
+    end
+    
+    def send_board_update_response(updated_game, connection, _params) do
+        body = Jason.encode!(updated_game)
+
         connection  
-        |> put_status(:ok)
-        |> send_resp(200, "ok")
+        |> put_status(200)
+        |> text(body)
     end
 
     def fetch_board_update_request(connection) do
         response = connection.body_params
         Map.fetch(response, "data")
     end
 
+    def list_to_map(list) do
```

Comment:
Why not just store the board as a map to begin with?

Reply:
This is one of my pain points. I'd like to deal with only maps since that's how I'm updating my board, by updating the map with the player token and cell number. However, the post request from JavaScript does not return a map, it returns a nested JSON object which needs to be converted to a map nested in a map while keeping the original keys from the nested array. I'm open to any ideas for this!

Reply:
I solved this. It was a simple fix of using an object instead of array on the frontend. I'll remove those map functions.